src/modules/collection/sagas.ts

```diff
@@ -180,23 +183,43 @@ export function* collectionSaga(legacyBuilderClient: BuilderAPI, client: Builder
   }
 
   function* handleFetchCollectionsRequest(action: FetchCollectionsRequestAction) {
-    const { address, params } = action.payload
+    const { address, params, useCachedResults } = action.payload
     try {
-      const response: PaginatedResource<Collection> | Collection[] = yield call([legacyBuilderClient, 'fetchCollections'], address, params)
-      if (isPaginated(response)) {
-        const { results, limit, page, pages, total } = response
-        yield put(
-          fetchCollectionsSuccess(results, {
-            limit,
-            page,
-            pages,
-            total
-          })
-        )
+      if (useCachedResults) {
+        const lastFetchParams: FetchCollectionsParams | undefined = yield select(getLastFetchParams)
+        if (equal(params, lastFetchParams)) {
+          const collections: Record<string, Collection> = yield select(getData)
+          const paginationData: CollectionPaginationData | null = yield select(getPaginationData)
+          yield put(
+            fetchCollectionsSuccess(Object.values(collections), paginationData ? toPaginationStats(paginationData) : undefined, params)
+          )
+        }
       } else {
-        yield put(fetchCollectionsSuccess(response))
+        const response: PaginatedResource<Collection> | Collection[] = yield call(
+          [legacyBuilderClient, 'fetchCollections'],
+          address,
+          params
+        )
+        if (isPaginated(response)) {
+          const { results, limit, page, pages, total } = response
+          yield put(
+            fetchCollectionsSuccess(
+              results,
+              {
+                limit,
+                page,
+                pages,
+                total
+              },
+              params
+            )
+          )
+        } else {
+          yield put(fetchCollectionsSuccess(response, undefined, params))
+        }
       }
     } catch (error) {
+      console.log('error: ', error)
```

Comment:
I think you might have left a console log here, right?

Reply:
removed!